nservicebus/azure-service-bus/configuration/configuration.md

```diff
@@ -0,0 +1,200 @@
+
+The topology will create the entities that it needs based on the following settings: 
+
+ * `Queues()`: Settings for queues.
+ * `Topics()`: Settings for topics.
+ * `Subscriptions()`: Settings for subscriptions.
+
+### Queues
+
+The following settings are available to define how queues should be created.
+
+ * `MaxSizeInMegabytes(SizeInMegabytes)`: The size of the queue, in megabytes.
+ * `MaxDeliveryCount(int)`: Sets the maximum delivery count, defaults to 10.
+ * `LockDuration(TimeSpan)`: The period of time that Azure Service Bus will lock a message before trying to redeliver it, defaults to 30 seconds.
+ * `ForwardDeadLetteredMessagesTo(string)`: Forward all dead lettered message to the specified entity.
+ * `ForwardDeadLetteredMessagesTo(Func<string, bool>, string)`: Forward all dead lettered message to the specified entity if the given condition is true (allows to exclude forwarding dead letters on the error queue for example).
+ * `DefaultMessageTimeToLive(TimeSpan)`: The maximum age of a message, defaults to TimeSpan.MaxValue.
+ * `EnableDeadLetteringOnMessageExpiration(bool)`: Messages that expire will be dead lettered, defaults to false.
+ * `EnableExpress(bool)`: Enables express mode, defaults to false.
+ * `EnableExpress(Func<string, bool>, string)`: Enables express mode when the given condition is true.
+ * `AutoDeleteOnIdle(TimeSpan)`: Auto deletes the entity if it hasn't been used for the given time frame.
+ * `EnablePartitioning(bool)`: Enables entity partitioning, defaults to false.
+ * `EnableBatchedOperations(bool)`: Enables server side batched operations, defaults to true.
+ * `RequiresDuplicateDetection(bool)`: Specifies whether the queue should perform duplicate detection, defaults to false.
+ * `DuplicateDetectionHistoryTimeWindow(TimeSpan)`: The time period in which duplicate detection should occur. 
+ * `SupportOrdering(bool)`: Best effort message ordering on the queue, defaults to false.
+ * `DescriptionFactory(Func<string, ReadOnlySettings, QueueDescription>)`: A factory method that allows to create a QueueDescription object from the Azure Service Bus SDK. Use this factory method to override any (future) setting that is not supported by the Queues API.
+
+### Topics
+ 
+The following settings are available to define how topics should be created.
+
+ * `MaxSizeInMegabytes(SizeInMegabytes)`: The size of the topic, in megabytes.
+ * `DefaultMessageTimeToLive(TimeSpan)`: The maximum age of a message, defaults to TimeSpan.MaxValue. 
+ * `AutoDeleteOnIdle(TimeSpan)`: Auto deletes the entity if it hasn't been used for the given time frame.
+ * `EnableBatchedOperations(bool)`: Enables server side batched operations, defaults to true.
+ * `EnableExpress(bool)`: Enables express mode, defaults to false.
+ * `EnableExpress(Func<string, bool>, bool)`: Enables express mode when the given condition is true.
+ * `EnableFilteringMessagesBeforePublishing(bool)`: Enables message filtering before they are published, defaults to false.
```

Comment:
This setting checks that there are subscribers before accepting a message on the topic (it will throw exceptions if there are no subscribers)